test/docker/aws/consumer_integration_test.go

```diff
@@ -0,0 +1,117 @@
+// +build integration
+
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+	patronSQS "github.com/beatlabs/patron/client/sqs"
+	sqsConsumer "github.com/beatlabs/patron/component/async/sqs"
+	"github.com/beatlabs/patron/correlation"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/opentracing/opentracing-go/mocktracer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type message struct {
+	ID string `json:"id"`
+}
+
+func Test_SQS_Consume(t *testing.T) {
+	const queueName = "test-sqs-consume"
+	const correlationID = "123"
+
+	api, err := createSQSAPI(runtime.getSQSEndpoint())
+	require.NoError(t, err)
+	queue, err := createSQSQueue(api, queueName)
+	require.NoError(t, err)
+
+	sent := sendMessage(t, api, correlationID, queue, "1", "2", "3")
+
+	mtr := mocktracer.New()
+	defer mtr.Reset()
+	opentracing.SetGlobalTracer(mtr)
+
+	factory, err := sqsConsumer.NewFactory(api, queueName)
+	require.NoError(t, err)
+	cns, err := factory.Create()
+	require.NoError(t, err)
+	ch, chErr, err := cns.Consume(context.Background())
+	require.NoError(t, err)
+
+	count := 0
+
+	chReceived := make(chan []*message)
+
+	go func() {
+		received := make([]*message, 0, len(sent))
+
+		for {
+			select {
```

Comment:
would it make sense to have also a 'timeout' ?

Reply:
besides the timeouts of the tests?

Reply:
yeah, there is a timeout in TestMain (if thats what you mean?) Just wondering if it would make sense to have timeout specific to each test.

Reply:
could make sense.. I am also using the CLI argument for timeout also... maybe we do not need to define another one?